source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.h

```diff
@@ -33,7 +35,8 @@ namespace Controller {
   GAUGE(gradient, NeverImport)                                                                     \
   GAUGE(min_rtt_calculation_active, Accumulate)                                                    \
   GAUGE(min_rtt_msecs, NeverImport)                                                                \
-  GAUGE(sample_rtt_msecs, NeverImport)
+  GAUGE(sample_rtt_msecs, NeverImport)                                                             \
+  COUNTER(empty_histogram)
```

Comment:
I just changed the stats name to be `race_found`. For this specific stats, my intention is that it is only used for test purpose, since we want to test the race condition in the unit test. From my perspective, [else](https://github.com/envoyproxy/envoy/pull/15004/files#diff-54be32369c917516472877e85b252da3b9fa60a3fc07a22c8cd2b65730a60097R122) code path will always have the possibility to hit as long as updateMinRTT() get invoked simultaneously.